app/controllers/api/v1/account_controller.rb

```diff
@@ -0,0 +1,20 @@
+# RESTful JSON API to retrieve data for My Account
+class Api::V1::AccountController < ApiController
+  before_action :authenticate_user!
+
+  def my_applications
+    applications = current_user_applications
+    listing_ids = applications.collect { |a| a['listingID'] }.uniq
+    listings = ListingService.listings(listing_ids.join(','))
+    applications.each do |app|
+      app['listing'] = listings.find { |l| l['listingID'] == app['listingID'] }
+    end
+    render json: { applications: applications }
+  end
```

Comment:
@chrisdolendo good call! yeah I didn't even think about the fact that we'd need to separately grab the listing data when first starting this feature, and I started doing it a couple different ways before settling on this one, but agree it could be refactored. Done! To please the private method gods and goddesses! 🙏  💃